modules/ocf_desktop/manifests/xsession.pp

```diff
@@ -72,6 +72,15 @@
     '/etc/lightdm/session-cleanup':
       mode    => '0755',
       source  => 'puppet:///modules/ocf_desktop/xsession/lightdm/session-cleanup';
+    '/opt/share/xsession/lightdm-gtk-greeter.po':
+      source  => 'puppet:///modules/ocf_desktop/xsession/lightdm/lightdm-gtk-greeter.po';
+  }
+
+  # overwrite greeter strings with OCF ones
+  exec { 'lightdm-greeter-compile-po':
+      command => 'msgcat /opt/share/xsession/lightdm-gtk-greeter.po | msgfmt -o \
+                    /usr/share/locale/en_US/LC_MESSAGES/lightdm-gtk-greeter.mo -',
+      require => File['/opt/share/xsession/lightdm-gtk-greeter.po'];
```

Comment:
you should put some kind of restriction on when this exec runs or it will run every time:
https://docs.puppetlabs.com/references/latest/type.html#exec-description

adding `refreshonly => true` and `subscribe => File['/opt/share/xsession/lightdm-gtk-greeter.po']` would work I think